### ISSUE TITLE
Enter 2FA input token before save model

### ISSUE THREAD

**Issue by jetwes** [laravel/fortify#47]
At the moment it is impossible to verify that the user activated 2fa in his device.
It should be mandatory that the user has to confirm the activation with an actual 2fa code.
This is not possible at the moment because in "Laravel\Fortify\Actions\EnableTwoFactorAuthentication" the database field is filled on activation. The generation of the secret has to be done before and the secret has to be an option in the action so it is possible to check a valid code before the secret is saved to the database.

related to [https://github.com/laravel/jetstream/issues/74](url)

**Comment by driesvints** [comment]
Gonna mark this as a duplicate of https://github.com/laravel/jetstream/issues/74 to keep the discussion focused.

**Comment by jetwes** [comment]
understood. But the implementation belongs to fortify in my opinion ;)

**Comment by jetwes** [comment]
ok - will look into this tonight
